src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/custom.py

```diff
@@ -1033,12 +1033,7 @@ def create_policy_assignment(cmd, policy=None, policy_set_definition=None,
     scope = _build_policy_scope(policy_client.config.subscription_id,
                                 resource_group_name, scope)
     policy_id = _resolve_policy_id(cmd, policy, policy_set_definition, policy_client)
-
-    if params:
-        if os.path.exists(params):
-            params = get_file_json(params)
-        else:
-            params = shell_safe_json_parse(params)
+    params = _load_file_string_or_uri(params, 'params', False)
```

Comment:
https://github.com/Azure/azure-cli/blob/51cc6713ea2147260447b3d32fe8dc7c7df3be01/src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/custom.py#L448

This error is incorrect as, near as I can tell, there are no `--foo/--foo-uri` parameters.

Reply:
You are correct.  I don't know what the original intention of this error msg was as I can't find any intended support for a special "-uri" suffix variant if the user wanted to explicitly indicate it was a URI.